controllers/multiclusterobservability/grafana.go

```diff
@@ -138,5 +125,15 @@ func GenerateGrafanaDataSource(
 		return &ctrl.Result{}, err
 	}
 
+	if !reflect.DeepEqual(grafanaDSFound.Data, dsSecret.Data) {
+		log.Info("Updating grafana datasource secret")
+		dsSecret.ObjectMeta.ResourceVersion = grafanaDSFound.ObjectMeta.ResourceVersion
```

Comment:
why we need to keep `ResourceVersion` without changes?

Reply:
maybe it will cause another reconcile?

Reply:
it's a no-harm change. 
for some resources, the update will fail if ResourceVersion is empty.